dotCMS/src/main/webapp/html/portlet/ext/contentlet/edit_contentlet_relationships.jsp

```diff
@@ -700,7 +702,9 @@
 								v = v.replace("\r\n","");
 								v = v.replace("\n","");
 								v = v.replace("\r","");
-								if(v.matches("(.*)-(.*)-(.*):(.*):(.*)")){
+								Pattern p = Pattern.compile("(.*)-(.*)-(.*):(.*):(.*)");

```

Comment:
It is great and might be we can encapsulate the logic on our StringUtils, such as:
```
private static Map<String, Pattern> patternCacheMap = new ConcurrentHashMap<>();

    /**
     * Returns true if the pattern match on the string
     * @param string  String string to search
     * @param pattern String pattern to search
     * @return boolean true if match
     */
    public static boolean matches (final String string, final String pattern) {

        if (!patternCacheMap.containsKey(pattern)) {

            patternCacheMap.put(pattern, Pattern.compile(pattern));
        }

        return patternCacheMap.get(pattern).matcher(string).matches();
    }
```
thx

Reply:
this may work, and I have a working case of this suggested implementation. the problem comes with the matching process, which depends on that nasty regex for date-time validation. compilation process is going to be cached but it may not work once a field with a huge amount of data is thrown at it.